examples/pytorch/pytorch_lightning_mnist.py

```diff
@@ -176,15 +175,27 @@ def test():
         setattr(model, 'train_dataloader', lambda: train_loader)
         setattr(model, 'val_dataloader', lambda: test_loader)
 
+        from pytorch_lightning.callbacks import Callback
+        class MyPrintingCallback(Callback):
+
+            def on_init_start(self, trainer):
+                print('Starting to init trainer!')
+
+            def on_init_end(self, trainer):
+                print('trainer is init now')
+
+            def on_train_end(self, trainer, pl_module):
+                print('do something when training ends')
+
+        callbacks = [MyPrintingCallback(), ModelCheckpoint(dirpath=ckpt_path)]
```

Comment:
Is the `chkpt with bcast` hanging still there? What's the difference of the code now?

Reply:
Ignore above, I think that's for estimator training.

Reply:
It works in this "pytorch_lightning_mnist.py", where estimator and petastorm loader is not used. 
It can be reproduced in "pytorch_lightning_spark_mnist.py " and the new test case that I added `test_model_checkpoint_callback` in this PR.